avocado/core/test.py

```diff
@@ -630,6 +630,12 @@ def get_state(self):
         state['class_name'] = self.__class__.__name__
         state['job_logdir'] = self.job.logdir
         state['job_unique_id'] = self.job.unique_id
+        params = getattr(self, 'params', None)
```

Comment:
Actually, getting a `None` here can be currently be considered an erroneous situation, because `params` is not an optional attribute or value of the `avocado.core.Test` class.   I recommend dropping this and the `if params is not None` check altogether.